### ISSUE TITLE
tiles not loading when a js-defined styles encounters an undefined property

### ISSUE THREAD

**Issue by burritojustice** [tangrams/tangram#653]
**TANGRAM VERSION:**

0.15.1

**ENVIRONMENT:**

Mac OS, Safari & Chrome

**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

Style a layer's parameter with a javascript function, and call on a geojson property than may not be in every feature

**RESULT:**

If a property that the function relies on is undefined, the resulting js error often kills the base tile (even if the data is in a layer that is separate from the base tile)

`Tangram v0.15.1 [error]: Error building tile mapzen/8/37/48/7: TypeError: Cannot read property 'length' of undefined`

```Tangram v0.15.1 [error]: Filter for this roads:shields:z-legit-shields:US-I:width_5char: `filter: function() { return (feature.shield_text.length === 5) }` failed with error 'Cannot read property 'length' of undefined', stack trace: TypeError: Cannot read property 'length' of undefined```

**EXPECTED RESULT:**

be more forgiving and continue to load the tile, even if a property is undefined in a particular feature

**Comment by matteblair** [comment]
For this particular circumstance you could rewrite your filter as:
```javascript
function() { return ((feature.shield_text || "").length === 5); }
```
or:
```yaml
filter:
  all:
    - shield_text: true
    - function() { return (feature.shield_text.length === 5) }
```

The code that seems to execute these filter functions seems to already use a try/catch block, so I'm confused why this would be halting processing for the whole tile 🤔 (https://github.com/tangrams/tangram/blob/master/src/styles/layer.js#L270-L278)

**Comment by burritojustice** [comment]
That would be good to find out why. I've learned to use `feature.value || ""` to get around it for data I'm importing, but a user will not always have control over the code in a style they are using, or the data in a tile...

**Comment by bcamper** [comment]
@burritojustice can you provide a scene and lat-lng that reliably fails?

**Comment by burritojustice** [comment]
that also fixes an example of tiles breaking around Anchorage that I just noticed:
https://tangrams.github.io/walkabout-style/#8.44166666666665/61.1693/-149.5973

http://tangrams.github.io/tangram-frame/?url=https://raw.githubusercontent.com/meetar/tangram/try-evalproperty/demos/scene.yaml&lib=https://rawgit.com/tangrams/tangram/try-evalproperty-build/dist/tangram.debug.js#8.44166666666665/61.1693/-149.5973

**Comment by bcamper** [comment]
Was released in v0.15.2
